.github/workflows/windows.yml

```diff
@@ -0,0 +1,25 @@
+name: Windows
+
+on: [push, pull_request]
+
+jobs:
+  ci:
+    runs-on: windows-2019
+    steps:
+      - name: Tell Git to use LF
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+      - uses: actions/checkout@v2
+      - name: Set up Cygwin
+        uses: egor-tensin/setup-cygwin@v2
+        with:
+          packages: git gcc-code make automake libiconv-devel libncurses-devel cygwin-devel glibc
+      - name: Test Tig
+        env:
+          CC: gcc
```

Comment:
Maybe setting the PATH environment variable could make a difference as appveyor.yml has a line with ` - cmd: SET PATH=C:\%TARGET%\bin;%PATH%`.

Reply:
Right, we actually need to prepend to the path to use the right binaries. I have made this work on a Windows system now. The tests work fine (but are really slow). I don't know a workaround for https://github.com/actions/runner/issues/241 that works on Windows, so I commented out `make test`. We can probably just leave it like that for now.

Reply:
Nice job. As we run the full test on Linux, it seems to be enough to just test the build on macOS and Windows. We can wait a feedback from @jonas to see if appveyor.yml can go as well.

Reply:
The Windows job is not very interesting because with Cygwin it's using the same GNU toolchain as Linux.
It is, however useful as documentation of how to quickly get Tig up and running on Windows. I'm no longer using the GitHub Actions library that abstract installing cygwin, because a single `choco install cygwin` is enough; that makes it easier to recreate locally as well.

Maybe I'll try to run tests on macOS as well (provided the `script` workaround to get a TTY works there).

Could also be interesting to test on different libcs and other LInux distribution (which would also document which packages to install). But that's all not super important I guess.

Reply:
We have seen in the past some Windows specific compilation issues (#858), so I think it is worth having the Windows job.
`script` should work on macOS but I don't know how we can run the tests with the Brew build.

Reply:
Adding this step to macos.yml should make Homebrew run the test suite for us:
```
    - name: Patch
      run: |
        cd "$(brew --repo homebrew/core)" && patch -p1 <<\EOF
diff --git a/Formula/tig.rb b/Formula/tig.rb
index 2e8c125660..f76d6bca1c 100644
--- a/Formula/tig.rb
+++ b/Formula/tig.rb
@@ -26,9 +26,11 @@ class Tig < Formula
   depends_on "readline"
 
   def install
+    ENV.deparallelize
     system "./autogen.sh" if build.head?
     system "./configure", "--prefix=#{prefix}", "--sysconfdir=#{etc}"
     system "make"
+    system "make", "test"
     # Ensure the configured `sysconfdir` is used during runtime by
     # installing in a separate step.
     system "make", "install", "sysconfdir=#{pkgshare}/examples"
EOF
```

Reply:
yep, that works